### PR TITLE
Add Arabic localization

### PR DESCRIPTION
**Pull request by Mahabmn**

### CODE REVIEW

**Review by daisukes — CHANGES_REQUESTED** [review]
Thank you for your PR to add Arabic localization!

I need to ask you to include a part of `NavCog3.xcodeproj` file.
Currently, added files cannot be seen on XCode. (Please check by cloning your master branch from github)
Please use `git add -p NavCog3.xcodeproj` command to carefully select the minimum required changes for adding the Arabic language.

**Comment by Mahabmn** [comment]
updated

File: NavCog3.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -467,6 +467,22 @@
...
 		D5D9DE6372649AF633976E29 /* libPods-NavCogFP.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; includeInIndex = 0; path = "libPods-NavCogFP.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC5E2875E17700A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC602875E81100A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC622875E82600A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC77287863C500A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC79287863D900A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:471]: please don't include this change

File: NavCog3.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -467,6 +467,22 @@
...
+		E2082D1E28B4B9F2008492D1 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Fingerprint.strings; sourceTree = "<group>"; };
+		E2301B7727FF74C800A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/HLPGeoJSON.strings; sourceTree = "<group>"; };
+		E2301B7827FF74D100A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/BlindView.strings; sourceTree = "<group>"; };
+		E2301B7927FF74E700A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.stringsdict; name = ar; path = ar.lproj/BlindView.stringsdict; sourceTree = "<group>"; };
+		E2301B7A27FF74F500A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Localizable.strings; sourceTree = "<group>"; };
+		E2301B7B27FF751E00A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/InfoPlist.strings; sourceTree = "<group>"; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:477]: looks good

File: NavCog3.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -467,6 +467,22 @@
...
+		E262DF9D28807750007BC3D5 /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:478]: don't include

File: NavCog3.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -467,6 +467,22 @@
...
+		E2997D0E29327D6800ABB819 /* dev.hashfile */ = {isa = PBXFileReference; fileEncoding = 4; indentWidth = 4; lastKnownFileType = text; path = dev.hashfile; sourceTree = "<group>"; tabWidth = 4; usesTabs = 0; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:479]: don't include

File: NavCog3.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -467,6 +467,22 @@
...
+		E201FC5E2875E17700A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC602875E81100A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC622875E82600A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC77287863C500A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E201FC79287863D900A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E2082D1E28B4B9F2008492D1 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Fingerprint.strings; sourceTree = "<group>"; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:472]: You may need to commit `ar.lproj/Fingerprint.strings` which is not included in this PR.

File: NavCog3.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -467,6 +467,22 @@
...
+		E2301B7A27FF74F500A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Localizable.strings; sourceTree = "<group>"; };
+		E2301B7B27FF751E00A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		E262DF9D28807750007BC3D5 /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
+		E2997D0E29327D6800ABB819 /* dev.hashfile */ = {isa = PBXFileReference; fileEncoding = 4; indentWidth = 4; lastKnownFileType = text; path = dev.hashfile; sourceTree = "<group>"; tabWidth = 4; usesTabs = 0; };
+		E2D0AAFE27FB893C0018AA63 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Main.strings; sourceTree = "<group>"; };
+		E2D0AAFF27FB893C0018AA63 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/LaunchScreen.strings; sourceTree = "<group>"; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:481]: looks good

File: NavCog3.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -467,6 +467,22 @@
...
+		E2D0AB0027FB893C0018AA63 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = SettingUtil/ar.lproj/HLPSettingView.strings; sourceTree = "<group>"; };
```
Comment (daisukes) [inline NavCog3.xcodeproj/project.pbxproj:482]: `SettingUtil/ar.lproj/HLPSettingView.strings` this file is not included in this PR

**Review by daisukes — CHANGES_REQUESTED** [review]
Hello @Mahabmn

Thank you for the commit!
The commit (https://github.com/hulop/NavCogIOSv3/pull/67/commits/234cc029d3b339048fa6064173d86fb3758b69e7) looks much better than the last commit (https://github.com/hulop/NavCogIOSv3/pull/67/commits/0e344aad429bca45f8bb5b03383ab5207d38e451) which includes lots of irrelevant changes.

I want a commit that includes only language localization related changes as I commented.

Please let me know if you have any question!

File: NavCog3.xcodeproj/project.pbxproj (hunk 7 above)
Comment (Mahabmn) [reply NavCog3.xcodeproj/project.pbxproj:482]: Done

File: NavCog3.xcodeproj/project.pbxproj (hunk 5 above)
Comment (Mahabmn) [reply NavCog3.xcodeproj/project.pbxproj:472]: done , but the file Is empty .

**Comment by Mahabmn** [comment]
Hello @daisukes 

I update the above comments, please let me know if there is any missing parts.

thanks

**Comment by daisukes** [comment]
@Mahabmn 

Could you revert your changes up to https://github.com/hulop/NavCogIOSv3/pull/67/commits/234cc029d3b339048fa6064173d86fb3758b69e7 ?
Then please follow my comments.

The current PR includes this commit (https://github.com/hulop/NavCogIOSv3/pull/67/commits/0e344aad429bca45f8bb5b03383ab5207d38e451) which has a lot of irrelevant changes. You should not include this change.

**Comment by Mahabmn** [comment]
Hello @daisukes

I update the above comments, please let me know if there is any missing parts.

thanks

**Comment by daisukes** [comment]
Hmm, I think it is difficult to fix the issue by uploading your project file.

I would do these commands where you are working on to revert to the commit `234cc029`
```
# move to the directory of repository
cd <your NavCogIOSv3 repository dir>

# checkout the target commit
git checkout 234cc029

# this command discard changes after `234cc029`
git checkout -B master
```

Then open `project.pbxproj` with a text editor (not XCode) and remove unrelated changes.
```
delete->	E201FC5E2875E17700A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
delete->	E201FC602875E81100A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
delete->	E201FC622875E82600A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
delete->	E201FC77287863C500A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
delete->	E201FC79287863D900A9ADDD /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
		E2082D1E28B4B9F2008492D1 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Fingerprint.strings; sourceTree = "<group>"; };
		E2301B7727FF74C800A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/HLPGeoJSON.strings; sourceTree = "<group>"; };
		E2301B7827FF74D100A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/BlindView.strings; sourceTree = "<group>"; };
		E2301B7927FF74E700A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.stringsdict; name = ar; path = ar.lproj/BlindView.stringsdict; sourceTree = "<group>"; };
		E2301B7A27FF74F500A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Localizable.strings; sourceTree = "<group>"; };
		E2301B7B27FF751E00A62C75 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/InfoPlist.strings; sourceTree = "<group>"; };
delete->	E262DF9D28807750007BC3D5 /* libPods-NavCogTool.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = "libPods-NavCogTool.a"; sourceTree = BUILT_PRODUCTS_DIR; };
delete->	E2997D0E29327D6800ABB819 /* dev.hashfile */ = {isa = PBXFileReference; fileEncoding = 4; indentWidth = 4; lastKnownFileType = text; path = dev.hashfile; sourceTree = "<group>"; tabWidth = 4; usesTabs = 0; };
		E2D0AAFE27FB893C0018AA63 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/Main.strings; sourceTree = "<group>"; };
		E2D0AAFF27FB893C0018AA63 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = ar.lproj/LaunchScreen.strings; sourceTree = "<group>"; };
		E2D0AB0027FB893C0018AA63 /* ar */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ar; path = SettingUtil/ar.lproj/HLPSettingView.strings; sourceTree = "<group>"; };
```

Then, use the following command to amend the commit
```
# stage your change
git add ./NavCog3.xcodeproj/project.pbxproj

# then make a commit
git commit -m "remove unrelated change"

# finally push with "-f" option. This option should be used with extra caution in general
git push -f
```

**Comment by daisukes** [comment]
If you have any questions/issues, please feel free to ask me!

**Comment by Mahabmn** [comment]
thank you , will work on it soon .

**Comment by Mahabmn** [comment]
Hello @daisukes 

sorry for the delay 

I have done the above updates, please let me know if I have to do any modifications. 

thank you

**Comment by daisukes** [comment]
@Mahabmn 

Thank you for the update! It looks much better than before, but I got the following error when I tried to build it.


```
error: Build input file cannot be found: '/Users/cabot/src/NavCogIOSv3-temp/NavCog3/SettingUtil/ar.lproj/HLPSettingView.strings'. Did you forget to declare this file as an output of a script phase or custom build rule which produces it? (in target 'NavCog3' from project 'NavCog3')
```

**Comment by Mahabmn** [comment]
hello 

I will check and update you

**Comment by Mahabmn** [comment]
Hello @daisukes 

Please try Now , I fix the issue . 

Thank you

**Comment by daisukes** [comment]
Great, now I can build with Arabic localization!

Is it okay that it still has English words?
You may need to use Japanese Main.strings file for Arabic translation. English file does not have all entries.

- Server Selection
  - (Ignore map title - server-side matter)
- Download
- (Disclaimer is also server-side matter, so please ignore)
- Mode Selection, Back button
- (Outdoor is also server-side)
- Do you use "NavCog" in Arabic?
- (LEGAL and Disclaimer is server-side)
- VERSION
- Route, From, To, Recent, clear, Route Options, Start, Preview
- maybe more


Simulator Screen Recording - iPhone 14 - 2023-04-12 at 08 37 27

**Comment by Mahabmn** [comment]
Hello 

will check the Japanese Main.strings and try to update the Arabic one .

**Review by daisukes — APPROVED** [review]
Looks good to me

**Comment by daisukes** [comment]
@Mahabmn I appreciate your support in adding the Arabic language! I will contact my team again about the server-side code merging status and let you know soon.

**Comment by Mahabmn** [comment]
Hello @daisukes 

thank you for your support during this process .

**Pull request merged**